packages/discovery-provider/plugins/pedalboard/apps/relay/src/middleware/antiAbuse.ts

```diff
@@ -28,23 +27,14 @@ export const antiAbuseMiddleware = async (
   next: NextFunction
 ) => {
   const aaoConfig = config.aao
-  const { ip, recoveredSigner: user } = response.locals.ctx
-  const decodedAbi = decodeAbi(
-    response.locals.ctx.validatedRelayRequest.encodedABI
-  )
-  const isUserCreate =
-    decodedAbi.action === 'Create' && decodedAbi.entityType === 'User'
-  const isUserDeactivate =
-    user.is_deactivated === false &&
-    decodedAbi.action === 'Update' &&
-    decodedAbi.entityType === 'User' &&
-    JSON.parse(decodedAbi.metadata).data.is_deactivated === true
-  // User creations must be allowed as AAO won't have the user yet,
-  // and deactivations must be allowed even for abuse.
-  if (isUserCreate || isUserDeactivate) {
+  const { ip, recoveredSigner, signerIsApp, createOrDeactivate} = response.locals.ctx
+
+  // no AAO to check and creates / deactivates should always be allowed
+  if (signerIsApp || createOrDeactivate) {
```

Comment:
AAO requires a handle so on creation this doesn't work, we also don't want AAO to be the reason someone can't deactivate their account